src/main/fc/init.c

```diff
@@ -753,16 +753,44 @@ void init(void)
     //The OSD need to be initialised after GYRO to avoid GYRO initialisation failure on some targets
 
     if (featureIsEnabled(FEATURE_OSD)) {
+        switch(osdConfig()->displayPortDevice) {
+
+        case OSD_DISPLAYPORT_DEVICE_NONE:
+            break;
+
+        case OSD_DISPLAYPORT_DEVICE_AUTO:
+            FALLTHROUGH;
+
+        case OSD_DISPLAYPORT_DEVICE_MAX7456:
 #if defined(USE_MAX7456)
-        // If there is a max7456 chip for the OSD then use it
-        osdDisplayPort = max7456DisplayPortInit(vcdProfile());
-#elif defined(USE_CMS) && defined(USE_MSP_DISPLAYPORT) && defined(USE_OSD_OVER_MSP_DISPLAYPORT) // OSD over MSP; not supported (yet)
-        osdDisplayPort = displayPortMspInit();
+            // If there is a max7456 chip for the OSD configured and detectd then use it.
+            osdDisplayPort = max7456DisplayPortInit(vcdProfile());
+            if (osdDisplayPort) {
+                break;
+            }
+            FALLTHROUGH;
+#else
+            break;
+#endif
+
+        case OSD_DISPLAYPORT_DEVICE_MSP:
+#if defined(USE_CMS) && defined(USE_MSP_DISPLAYPORT) && defined(USE_OSD_OVER_MSP_DISPLAYPORT)
+            osdDisplayPort = displayPortMspInit();
+            if (osdDisplayPort) {
+                break;
+            }
+            FALLTHROUGH;
+#else
+            break;
 #endif
```

Comment:
I think it would be easier to read if you used the following construct:

```
#if defined(USE_MAX7456)
        case OSD_DISPLAYPORT_DEVICE_MAX7456:
￼            // If there is a max7456 chip for the OSD configured and detectd then use it.
￼            osdDisplayPort = max7456DisplayPortInit(vcdProfile());
￼            if (osdDisplayPort) {
￼                break;
￼            }
￼
￼            break;
￼#endif
        default:

        break;
```

Reply:
`switch` on `osdDisplayPortDevice_e` doesn't like unhandled enum element. What do you think?

Reply:
Isn't that caught by the `default`?